packages/clerk-js/src/ui/elements/UserPreview.tsx

```diff
@@ -77,39 +77,41 @@ export const UserPreview = (props: UserPreviewProps) => {
       sx={[{ minWidth: '0px', width: '100%' }, sx]}
       {...rest}
     >
-      {showAvatar ? (
-        <Flex
-          elementDescriptor={descriptors.userPreviewAvatarContainer}
-          elementId={descriptors.userPreviewAvatarContainer.setId(elementId)}
-          justify='center'
-          sx={{ position: 'relative' }}
-        >
-          <UserAvatar
-            boxElementDescriptor={descriptors.userPreviewAvatarBox}
-            imageElementDescriptor={descriptors.userPreviewAvatarImage}
-            {...user}
-            {...externalAccount}
-            {...samlAccount}
-            name={name}
-            avatarUrl={imageUrl}
-            size={getAvatarSizes}
-            sx={avatarSx}
-            rounded={rounded}
-          />
+      {imageUrl ? (
```

Comment:
This is necessary, because 

1. We don't want to display the elements if imageUrl does not exist as the html element will reserve space
2. We cannot use `showAvatar` because it will still an element that just reserves height (necessary for switcher)

Added a comment in the code, as to why this is needed